src/observer.ts

```diff
@@ -131,9 +134,18 @@ export class DataLayerObserver {
     }
 
     if (rules) {
+      const ruleRegistrationSpan = Telemetry.startSpan(telemetryType.ruleRegistrationSpan);
       rules.forEach((rule: DataLayerRule) => this.registerRule(rule));
+      // TODO(nate): Remove this call when the record log level is deprecated. Removing breaks tests
+      // so there may be some test state management issues to address
       Logger.getInstance().record('DLO rule count', { numericValue: rules.length });
```

Comment:
Per the comment, removing this logging statement causes test failures. I suspect there's some global test state that isn't being handled properly. Rather than resolving in this PR which already has a decent sized scope, I plan to address this when we deprecate the `record` log level.